static/app/views/replays/detail/header/replayDetailsUserBadge.tsx

```diff
@@ -25,6 +26,31 @@ export default function ReplayDetailsUserBadge({readerResult}: Props) {
   const replayRecord = readerResult.replayRecord;
   const replay = readerResult.replay;
 
+  const [isLive, setIsLive] = useState(replay?.getIsLive());
+
+  const computeIsLive = useCallback(() => replay?.getIsLive(), [replay]);
+
+  useEffect(() => {
+    // Immediately update if props change
+    setIsLive(computeIsLive());
```

Comment:
So we call `computeIsLive` 4 times in here, I don't know how complex `replay.getIsLive` is, but we could save the return value to a local var so we'd only call it once.

But that'd also be wrong since we need a freshly computed value inside of the `setInterval`. I could see someone refactoring and mistakenly doing this. I'd recommend having one var inside of the interval, and maybe one outside, and add some comments inside of the `setInterval`.